typescript/packages/plugins/ionic/src/ionic.service.ts

```diff
@@ -0,0 +1,144 @@
+import { Tool } from "@goat-sdk/core";
+import { EVMTransactionOptions, EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { z } from "zod";
+import { parseUnits, formatUnits, type Address, erc20Abi } from "viem";
+import type { HealthMetrics, PoolData } from "./types";
+import { ionicProtocolAddresses } from "./ionic.config";
+
+import { poolLensAbi } from "./abis/PoolLens";
+import { comptrollerAbi } from "./abis/Comptroller"; 
+import { ComptrollerProxyAbi } from "./abis/ComptrollerProxyAbi";
+import { poolAbi } from "./abis/Pool";
+import { poolDirectoryAbi } from "./abis/PoolDirectory";
+
+import {
+  SupplyAssetParameters,
+  BorrowAssetParameters,
+  GetHealthMetricsParameters
+} from "./parameters";
+
+export class IonicService {
+  private supportedTokens: string[];
+
+  constructor(supportedTokens?: string[]) {
+    this.supportedTokens = supportedTokens || Object.keys(ionicProtocolAddresses[34443].assets);
+  }
+
+  private async getAssetConfig(chainId: number, symbol: string): Promise<{ address: Address, decimals: number }> {
+    const config = ionicProtocolAddresses[chainId]?.assets?.[symbol];
+    if (!config?.address || config.decimals === undefined) {
+      throw new Error(`Asset ${symbol} not found in Ionic Protocol addresses for chain ${chainId}`);
+    }
+    return { address: config.address as Address, decimals: config.decimals };
+  }
+
+  @Tool({
+    name: "ionic_supply_asset",
+    description: "Supply an asset to an Ionic Protocol pool"
+  })
+  async supplyAsset(
+    wallet: EVMWalletClient,
+    params: SupplyAssetParameters,
+  ): Promise<void> {
+    const { asset, amount } = params;
+    const chain = await wallet.getChain();
+
+    // approval is handled by the erc20 plugin
+
+    // Get token address from config
+    const tokenAddress = ionicProtocolAddresses[chain.id]?.assets?.[asset]?.address;
+    if (!tokenAddress) {
+      throw new Error(`Token address not found for ${asset} on chain ${chain.id}`);
+    }
+
+    
+    // Then supply to pool
+    await wallet.sendTransaction({
+      to: tokenAddress,
+      abi: poolAbi,
+      functionName: "mint",
+      args: [BigInt(amount)]
+    });
```

Comment:
You should return the tx hash at least, if not, how do you know this worked?

Reply:
I got to know it worked by checking my onchain account.